options/options.go

```diff
@@ -1291,6 +1269,9 @@ type msgLiteralResolver struct {
 }
 
 func (r *msgLiteralResolver) FindMessageByName(message protoreflect.FullName) (protoreflect.MessageType, error) {
+	if r.interp.resolver == nil {
```

Comment:
Why is this check needed?

Reply:
Sorry, I forgot to call attention to this in the PR description. While I was making these changes, I noticed several places that used `interpreter.resolver` without checking for nil. It's not nil in practice in the tests, but there is an entry point where it can be nil, which could then lead to panics (the `resolver` field is an interface, so calling a method on a nil value will panic).

So I updated the code to properly check for nil everywhere while I was in here. Those are the changes at lines 708, 1272 (here), 1292, and 1316.